CI.hs

```diff
@@ -176,6 +176,33 @@ parseStackOptions = StackOptions
        <> Opts.help "If specified, pass '--ghc-options=\"xxx\"' to stack"
         ))
 
+-- [Note : GHC now depends on exceptions package]
+-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+-- As of
+-- https://gitlab.haskell.org/ghc/ghc/-/commit/30272412fa437ab8e7a8035db94a278e10513413
+-- (4th May 2020), certain GHC modules depend on the exceptions
+-- package. Depending on the boot compiler, this package may or may
+-- not be present and if it's missing, determining the list of
+-- ghc-lib-parser modules (via 'calcParserModules') will fail. The
+-- function 'appyPatchHadrianStackYaml' guarantees it is available
+-- if its needed.
+
+-- Patch Hadrian's Cabal.
+-- TODO (SF, 2020-09-20): Move this function and the above note to
```

Comment:
Any reason not do this now? This seems pretty crucial. We don’t want to force people to use `CI`. `ghc-lib-gen` should work on its own.

Reply:
there's no deep reason just that i spotted in the course of this work it's in the wrong place. sure, let's knock it over. stand by.

Reply:
done